test/utils/daemonsets/daemonsets.go

```diff
@@ -0,0 +1,98 @@
+package daemonsets
+
+import (
+	v1 "k8s.io/api/apps/v1"
+	v1core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func CreateDaemonSetsTemplate(dsName, namespace, containerName, imageWithVersion string) *v1.DaemonSet {
+
+	dsAnnotations := make(map[string]string)
+	dsAnnotations["debug.openshift.io/source-container"] = containerName
+	dsAnnotations["openshift.io/scc"] = "node-exporter"
+	matchLabels := make(map[string]string)
+	matchLabels["name"] = dsName
+
+	var trueBool bool = true
+	var zeroInt int64 = 0
+	var zeroInt32 int32 = 0
+	var preempt = v1core.PreemptLowerPriority
+	var tolerationsSeconds int64 = 300
+	var hostPathType = v1core.HostPathDirectory
+
+	container := v1core.Container{
+		Name:            containerName,
+		Image:           imageWithVersion,
+		ImagePullPolicy: "Always",
+		Command:         []string{"/bin/sh"},
+		SecurityContext: &v1core.SecurityContext{
+			Privileged: &trueBool,
+			RunAsUser:  &zeroInt,
+		},
+		Stdin:                  true,
+		StdinOnce:              true,
+		TerminationMessagePath: "/dev/termination-log",
+		TTY:                    true,
+		VolumeMounts: []v1core.VolumeMount{
+			{
+				MountPath: "/host",
+				Name:      "host",
+			},
+		},
+	}
+
+	return &v1.DaemonSet{
+
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        dsName,
+			Namespace:   namespace,
+			Annotations: dsAnnotations,
+		},
+		Spec: v1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: matchLabels,
+			},
+			Template: v1core.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: matchLabels,
+				},
+				Spec: v1core.PodSpec{
+					Containers:       []v1core.Container{container},
+					PreemptionPolicy: &preempt,
+					Priority:         &zeroInt32,
+
+					Tolerations: []v1core.Toleration{
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/not-ready",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/unreachable",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect: "NoSchedule",
+							Key:    "node-role.kubernetes.io/master",
```

Comment:
Will this be an issue with the Single node, since the node will have a master label?

Reply:
I have not tested with SNO yet - I would get back once that is done. We have other issues in the test cluster and hence  there is a delay.

Reply:
you could remove that when SNO detected